riverdriver/riverpgxv5/river_pgx_v5_driver.go

```diff
@@ -38,10 +44,13 @@ func New(dbPool *pgxpool.Pool) *Driver {
 	return &Driver{dbPool: dbPool, queries: dbsqlc.New()}
 }
 
-func (d *Driver) GetDBPool() *pgxpool.Pool                      { return d.dbPool }
-func (d *Driver) GetExecutor() riverdriver.Executor             { return &Executor{d.dbPool, dbsqlc.New()} }
-func (d *Driver) UnwrapExecutor(tx pgx.Tx) riverdriver.Executor { return &Executor{tx, dbsqlc.New()} }
-func (d *Driver) UnwrapTx(tx pgx.Tx) pgx.Tx                     { return tx }
+func (d *Driver) GetExecutor() riverdriver.Executor { return &Executor{d.dbPool, dbsqlc.New()} }
+func (d *Driver) GetListener() riverdriver.Listener { return &Listener{dbPool: d.dbPool} }
+func (d *Driver) HasPool() bool                     { return d.dbPool != nil }
+
+func (d *Driver) UnwrapExecutor(tx pgx.Tx) riverdriver.ExecutorTx {
+	return &ExecutorTx{Executor: Executor{tx, dbsqlc.New()}, tx: tx}
```

Comment:
So I kinda like the Go default where it'll let you get away with for structs within the same package, but it's an error doing it across packages.

It's definitely a stylistic thing though admittedly. I tend to only do it when there's just a couple fields and the struct definition is closeby. I'm generally not super concerned about it because it's hard to misuse — you can't miss a field or assign a field of the wrong type. I could probably be convinced to stop using it though.